### PR TITLE
feat: adding python cac and experimentation client wrapper

### PR DESCRIPTION
**Pull request by namitgoel**
## Problem
Describe the problem you are trying to solve here

## Solution
Provide a brief summary of your solution so that reviewers can understand your code

## Environment variable changes

What ENVs need to be added or changed

## Pre-deployment activity
Things needed to be done before deploying this change (if any)

## Post-deployment activity
Things needed to be done after deploying this change (if any)

## API changes
| Endpoint        | Method           | Request body  | Response Body |
| ------------- |:-------------:| -----:| ----------------:|
| API      | GET/POST, etc | request | response |

## Possible Issues in the future
Describe any possible issues that could occur because of this change

### CODE REVIEW

File: clients/python/cacclient/client.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            print("Some Error Occur while creating new client ", error_message)
+        return resp
+
+    def start_cac_polling_update(self):
+        threading.Thread(target=self._polling_update_worker).start()
+
+    def _polling_update_worker(self):
+        self.rust_lib.cac_start_polling_update(self.tenant.encode())
+
+    def get_cac_config(self, filter_query: str, filter_prefix: str) -> str:
```
Comment (pratikmishra356) [inline clients/python/cacclient/client.py:83]: @namitgoel filter_query, filter_prefix is optional argument , can you make the corresponding changes

**Review by pratikmishra356 — CHANGES_REQUESTED** [review]

File: clients/python/cacclient/client.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    def free_cac_client(self, client_ptr: str):
+        self.rust_lib.cac_free_client(client_ptr.encode())
+
+    def free_cac_string(self, string: str):
+        self.rust_lib.cac_free_string(string.encode())
+
+    def get_last_modified(self) -> str:
+        return self.rust_lib.cac_get_last_modified(self.get_cac_client()).decode()
+
+    def get_resolved_config(self, query: str, filter_keys: str, merge_strategy: str) -> str:
```
Comment (pratikmishra356) [inline clients/python/cacclient/client.py:96]: here filter keys is options

**Review by pratikmishra356 — CHANGES_REQUESTED** [review]

File: clients/python/cacclient/client.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        return self.rust_lib.cac_get_resolved_config(
+            self.get_cac_client(), query.encode(), filter_keys.encode(), merge_strategy.encode()).decode()
+
+    def get_default_config(self, filter_keys: str) -> str:
```
Comment (pratikmishra356) [inline clients/python/cacclient/client.py:100]: here also it is optional

**Review by pratikmishra356 — CHANGES_REQUESTED** [review]

File: clients/python/cacclient/client.py (hunk 1 above)
Comment (namitgoel) [reply clients/python/cacclient/client.py:83]: @pratikmishra356 done

File: clients/python/cacclient/client.py (hunk 3 above)
Comment (namitgoel) [reply clients/python/cacclient/client.py:100]: @pratikmishra356 done

File: clients/python/cacclient/client.py (hunk 2 above)
Comment (namitgoel) [reply clients/python/cacclient/client.py:96]: @pratikmishra356 done

File: clients/python/cacclient/client.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            print("Some Error Occur while creating new client ", error_message)
+        return resp
+
+    def start_cac_polling_update(self):
+        threading.Thread(target=self._polling_update_worker).start()
+
+    def _polling_update_worker(self):
+        self.rust_lib.cac_start_polling_update(self.tenant.encode())
+
+    def get_cac_config(self, filter_query: str = "{}", filter_prefix: str = "") -> str:
```
Comment (pratikmishra356) [inline clients/python/cacclient/client.py:83]: @namitgoel can we add filter_prefix: str = None, default as None instead of empty string in all arguments

**Review by pratikmishra356 — CHANGES_REQUESTED** [review]

File: clients/python/expclient/client.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        return self.rust_lib.expt_get_filtered_satisfied_experiments(
+            self.get_experimentation_client(), context.encode(), filter_prefix.encode()
+        ).decode()
+
+    def get_applicable_variant(self, context: str, toss: int) -> str:
+        return self.rust_lib.expt_get_applicable_variant(
+            self.get_experimentation_client(), context.encode(), toss
+        ).decode()
+
+    def get_satisfied_experiments(self, context: str, filter_prefix: str) -> str:
```
Comment (pratikmishra356) [inline clients/python/expclient/client.py:104]: here filter_prefix is also optional

**Review by pratikmishra356 — CHANGES_REQUESTED** [review]

File: clients/python/expclient/client.py (hunk 5 above)
Comment (namitgoel) [reply clients/python/expclient/client.py:104]: @pratikmishra356 done

**Review by pratikmishra356 — APPROVED** [review]

**Review by Datron — APPROVED** [review]
@namitgoel can we include function documentation in client-context-aware-configuration.md too? Not a blocker

**Pull request merged**